### PR TITLE
Fix indentation when using inline vault encrypt

### PR DESCRIPTION
**Pull request by jeinwag**
This should fix the non-perfect indentation when using inline encrypt.

### CODE REVIEW

File: src/features/vault.ts [hunk 1]
```diff
@@ -92,8 +92,11 @@ export const toggleEncrypt = async (): Promise<void> => {
         editBuilder.replace(
           selection,
           encryptedText.replace(
-            /\n/g,
-            `\n${' '.repeat(selection.start.character)}`
+            /\n\s*/g,
+            `\n${' '.repeat(
+              (getIndentationLevel(editor, selection) + 1) *
+                Number(editor.options.tabSize)
+            )}`
```
Comment (webknjaz) [inline src/features/vault.ts:99]: This is quite complex. Maybe move it out and assign it to a var?

File: src/features/vault.ts [hunk 2]
```diff
@@ -281,3 +284,19 @@ const execCwd = (cmd: string, cwd: string | undefined) => {
...
   return exec(cmd, { cwd: cwd });
 };
+
+const getIndentationLevel = (
+  editor: vscode.TextEditor,
+  selection: vscode.Selection
+): number => {
+  if (!editor.options.tabSize) {
+    // according to VS code docs, tabSize is always defined when getting options of an editor
+    throw 'tabSize undefined, this should never happen';
```
Comment (webknjaz) [inline src/features/vault.ts:294]:
```suggestion
    throw new Error('The `tabSize` option is not defined, this should never happen.');
```

File: src/features/vault.ts [hunk 3]
```diff
@@ -281,3 +284,19 @@ const execCwd = (cmd: string, cwd: string | undefined) => {
...
+  }
+  const startLine = editor.document.lineAt(selection.start.line).text;
+  const indentationMatches = startLine.match(/^\s*/);
+  const leadingWhitespaces = !!indentationMatches
+    ? indentationMatches[0].length
+    : 0;
```
Comment (webknjaz) [inline src/features/vault.ts:300]: Could this be something like
```suggestion
  const leadingWhitespaces = indentationMatches?[0].?length || 0;
```
?

File: src/features/vault.ts [hunk 4]
```diff
@@ -281,3 +284,19 @@ const execCwd = (cmd: string, cwd: string | undefined) => {
...
+  return leadingWhitespaces / Number(editor.options.tabSize);
```
Comment (webknjaz) [inline src/features/vault.ts:301]: Should this be
```suggestion
  return leadingWhitespaces / new Number(editor.options.tabSize);
```

Comment (webknjaz) [reply src/features/vault.ts:301]: Ah, never mind. The docs seem to imply that it's not a constuctor.

File: src/features/vault.ts (hunk 1 above)
Comment (jeinwag) [reply src/features/vault.ts]: Done.

**Review by ssbarnea — CHANGES_REQUESTED** [review]
Apparently it is not working as intended because the indentation level of the encrypted text needs to be one level down from the line with !vault marker.

Keep in mind that original line can be at any indentation level.

```
  vars:
    username: !vault |
    $ANSIBLE_VAULT;1.1;AES256
    ...
    password:
      xxx: !vault |
     $ANSIBLE_VAULT;1.1;AES256
     ....
```

If you looked at the example above, for the first case the indentation was 4 chars instead of 6 and for the second case it was 5 chars instead of of expected 8.

My expectation is that the encrypted text to be one level down (2 more chars) than the vault. Am I wrong about it?

This was result of testing code from the change (or at least I hope that is what I did).

**Comment by jeinwag** [comment]
@ssbarnea sorry, my last commit broke it. I fixed it again.

**Review by ssbarnea — APPROVED** [review]
Good that I tested. Indeed the last version works as expected. I personally find the popup notification message distracting but changing it would be outside the scope of this change. I will see if other users find it the same or not.

**Pull request merged**